### DISCUSSION TITLE
How to handle Unknown Prop warning on aria/button for `onPress`?

### DISCUSSION THREAD

**Discussion by PixeledCode** [adobe/react-spectrum#3666] [Q&A] [answered]
So, I am getting this warning on using react-aria/button `Warning: Unknown event handler property onPress. It will be ignored.`

As far as I am aware, it does not have any side-effect, but I still would like to remove this warning on every button used. 
The closest issue I found was this: #688

Any help is appreciated.

**Comment by ktabors** [comment]
Can you provide a CodeSandbox of the code that is causing this warning? How is it different from the examples in the docs?

**Comment by reidbarber** [comment, marked as answer]
You're probably spreading all the props down to the button element. You'll want to spread only `buttonProps`:

```jsx
import {useButton} from 'react-aria';
import {useRef} from 'react';

function Button(props) {
  let ref = useRef();
  let { buttonProps } = useButton(props, ref);
  let { children } = props;

  return (
    <button {...buttonProps} ref={ref}>
      {children}
    </button>
  );
}
```

or be intentional with what get's passed down so that it's only valid DOM props.

**Reply by PixeledCode** [reply]
It was this issue. I had other props as well so now I am intentional including them and not spreading all of the props. Thanks!
